python/pyspark/pandas/frame.py

```diff
@@ -375,6 +373,17 @@ class DataFrame(Frame, Generic[T]):
     copy : boolean, default False
         Copy data from inputs. Only affects DataFrame / 2d ndarray input
 
+    Notes
+    -----
+    Since 3.4.0, it deals with `index` in this way:
```

Comment:
Can we use `.. versionchanged:: 3.4.0` instead if it describes about the different behavior with previous version  ??